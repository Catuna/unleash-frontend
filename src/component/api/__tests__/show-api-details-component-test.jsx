import React from 'react';

import ShowApiDetailsComponent from '../show-api-details-component';
import renderer from 'react-test-renderer';

jest.mock('react-mdl');

test('renders correctly with empty api details', () => {
    const tree = renderer.create(<ShowApiDetailsComponent fetchAll={jest.fn()} apiDetails={{}} />).toJSON();
    expect(tree).toMatchSnapshot();
});

test('renders correctly with details', () => {
    const tree = renderer
        .create(<ShowApiDetailsComponent fetchAll={jest.fn()} apiDetails={{ version: '1.1.0' }} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
