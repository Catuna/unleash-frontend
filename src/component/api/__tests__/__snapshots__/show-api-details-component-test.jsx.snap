// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with details 1`] = `
<react-mdl-FooterSection
  logo="Unleash 1.1.0"
  type="bottom"
>
  <react-mdl-FooterLinkList>
    <a
      href="https://github.com/Unleash/unleash/"
      target="_blank"
    >
      GitHub
    </a>
    <a
      href="https://www.finn.no"
      target="_blank"
    >
      <small>
        A product by
      </small>
       FINN.no
    </a>
  </react-mdl-FooterLinkList>
</react-mdl-FooterSection>
`;

exports[`renders correctly with empty api details 1`] = `
<react-mdl-FooterSection
  logo="Unleash "
  type="bottom"
>
  <react-mdl-FooterLinkList>
    <a
      href="https://github.com/Unleash/unleash/"
      target="_blank"
    >
      GitHub
    </a>
    <a
      href="https://www.finn.no"
      target="_blank"
    >
      <small>
        A product by
      </small>
       FINN.no
    </a>
  </react-mdl-FooterLinkList>
</react-mdl-FooterSection>
`;
